nircam_simulator/scripts/obs_generator.py

```diff
@@ -1366,11 +1366,29 @@ def calcFrameTime(self, frame):
         yd, xd = frame.shape
         #self.frametime = (xd/self.params['Readout']['namp'] + 12.) * (yd+1) * 10.00 * 1.e-6
         #UPDATED VERSION, 16 Sept 2017
-        colpad = 12
-        rowpad = 2
-        if ((xd <= 8) & (yd <= 8)):
-            rowpad = 3
-        self.frametime = ((1.0 * xd/self.params['Readout']['namp'] + colpad) * (yd+rowpad)) * 1.e-5
+        if 'nircam' in self.params['Inst']['instrument']:
+          colpad = 12
+          rowpad = 2
+          if ((xd <= 8) & (yd <= 8)):
+              rowpad = 3
+          self.frametime = ((1.0 * xd / self.params['Readout']['namp'] + colpad) * (yd + rowpad)) * 1.e-5
+        else:
```

Comment:
Same as above, replace by
`elif self.params['Inst']['instrument'].lower() in ['niriss', 'fgs']:`